core/chains/evm/gas/chainoracles/models.go

```diff
@@ -0,0 +1,18 @@
+package chainoracles
+
+import (
+	"context"
+
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+	"github.com/smartcontractkit/chainlink/v2/core/services"
+)
+
+// L1Oracle provides interface for fetching L1-specific fee components if the chain is an L2.
+// For example, on Optimistic Rollups, this oracle can return rollup-specific l1BaseFee
+//
+//go:generate mockery --quiet --name L1Oracle --output ./mocks/ --case=underscore
+type L1Oracle interface {
```

Comment:
What about renaming to UnderlyingL1.

Reply:
I would prefer `L1Oracle` since it is not actually calling the underlying L1, it is fetching from the L1 oracle contract provided by the L2